### PR TITLE
perf: Cache resource keys of resw

### LINKED ISSUE #9844: XAML Generator should cache parsed resw resources
## What would you like to be added:

This method should use a caching technique similar to https://github.com/unoplatform/uno/pull/9727: 
https://github.com/unoplatform/uno/blob/bcf66abbaa2e55f018aed75e0af3df58fbb9adb6/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs#L600

## Why is this needed:

Improved generation performance

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #9844

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/ByteSequenceComparer.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Uno.UI.SourceGenerators.XamlGenerator;
+
+internal static class ByteSequenceComparer
+{
+	// https://github.com/dotnet/roslyn/blob/6f47a0b611ab83ad1b7f8b404cc74098d37e283d/src/Compilers/Core/Portable/Collections/ByteSequenceComparer.cs#L24-L45
+	internal static bool Equals(ImmutableArray<byte> x, ImmutableArray<byte> y)
```
Comment (jeromelaban) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/ByteSequenceComparer.cs:13]: After thinking about it, it should go into the Uno.Extensions folder as well.

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.ResourceCacheKey.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+using Uno.Extensions;
+
+namespace Uno.UI.SourceGenerators.XamlGenerator;
+
+internal partial class XamlCodeGeneration
+{
+	private static readonly ConcurrentDictionary<ResourceCacheKey, CachedResource> _cachedResources = new();
+	private static readonly TimeSpan _cacheEntryLifetime = new TimeSpan(hours: 1, minutes: 0, seconds: 0);
+
+	private static void ClearCache()
```
Comment (jeromelaban) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.ResourceCacheKey.cs:18]:
```suggestion
	private static void ScavengeCache()
```

**Review by jeromelaban — APPROVED** [review]

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/ByteSequenceComparer.cs (hunk 1 above)
Comment (Youssef1313) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/ByteSequenceComparer.cs:13]: This one C:\Users\PC\Desktop\uno\src\Uno.Foundation\Uno.Core.Extensions\Uno.Core.Extensions.Equality ?

Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/ByteSequenceComparer.cs:13]: Sure!

**Review by jeromelaban — APPROVED** [review]

**Comment by Youssef1313** [comment]
@jeromelaban Is this ready to merge?

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**